src/modules/snd/libkvisnd.cpp

```diff
@@ -397,22 +397,26 @@ bool KviSoundPlayer::play(const QString & szFileName)
 
 	if(!e)
 	{
-		if(
-		    (!KVI_OPTION_STRING(KviOption_stringSoundSystem).isEmpty()) &&
-		    (!KviQString::equalCI(KVI_OPTION_STRING(KviOption_stringSoundSystem), "unknown")) &&
-		    (!KviQString::equalCI(KVI_OPTION_STRING(KviOption_stringSoundSystem), "null")))
-		{
-			qDebug(
-			    "Sound system '%s' is not valid, you may want to re-configure it in the options dialog...",
-			    KVI_OPTION_STRING(KviOption_stringSoundSystem).toUtf8().data());
-			return false; // detection already attempted (and failed?)
-		}
+		qDebug(
+		   "Sound system '%s' is not valid, attempting autodetect...",
```

Comment:
I've tested it, and it somewhat works. I was able to use "/play" command, but if you go to settings, it shows "null" there in sound system.

However, this is a different "null" from the one set explicitly - "/play" didn't work after explicit setting of system to null, but it worked (and autodetected qt) when I did /play without explicitly setting system to null first

Reply:
Would it make more sense to do the autodetection on startup instead when the old sound system is invalid? Let's not touch "null" to make it easy to disable sounds altogether

Reply:
> However, this is a different "null" from the one set explicitly - "/play" didn't work after explicit setting of system to null, but it worked (and autodetected qt) when I did /play without explicitly setting system to null first

I guess the first time KviOption_stringSoundSystem was still set to "phonon"m but your kvirc binary didn't support it, so the "phonon" option dialog was missing and the default "null" was selected.
If you set "null" explicitly, kvirc won't force an autodetect.

> Would it make more sense to do the autodetection on startup instead when the old sound system is invalid?

kvirc startup or snd module startup? Both seems unfair to me, because they would force change an option even if the user doesn't want/need to, eg  `echo $snd.isMuted`

Reply:
> kvirc startup or snd module startup? Both seems unfair to me, because they would force change an option even if the user doesn't want/need to, eg echo $snd.isMuted

Yeah, good point, not sure.

> so the "phonon" option dialog was missing and the default "null" was selected.

Maybe then make this option more clear between the two "null"s - one is "null", another is "invalid, will autodetect", or something like that?

Reply:
That would mean adding a fake option to the dropdownlist, adding some code to detect if the user chose that fake option, filtering it out to avoid it getting in the config file, etc..
I see no gain for the user.

Reply:
Ok, what's more confusing?

1) changing the option during `echo $snd.isMuted`

or

2) having a fake null in options which looks indistinguishable from the real null, but behaves differently - this is the current situation

I think I'd prefer 1) over 2)

Reply:
So, I've tried to add fake option to the settings:

![immagine](https://github.com/user-attachments/assets/71f47a06-e14b-435e-bd1c-dea01fafbba1)

This worked, but pressing the "test" button would call snd.play with an invalid sound system, forcing an autodetect that would play the sound but keep the dummy option selected. So it was even more confusing :)

At the end, in [1b045ba](https://github.com/kvirc/KVIrc/pull/2669/commits/1b045baabfdfd608023b42c022cd693647dbf078) the sound options dialog was changed to force an autodetect if the sound system defined in the options is invalid.

Reply:
Thanks!